### PR TITLE
frontend-feat: #5395 Grouped notifications in both customer and driver side

### PR DESCRIPTION
**Pull request by spravinkumar9952**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

classified notifications into groups.
[video](https://drive.google.com/file/d/1II5A22wKclyDKpcVEeqJh-zvFOXIBhx0/view?usp=sharing)

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 1]
```diff
@@ -312,8 +315,13 @@ else if (notificationDeepLinkVector != null) {
         initApp(viewParam, deepLinkJson);
 
         mFirebaseAnalytics.logEvent(isMigrated ?"migrate_local_store_success" : "migrate_local_store_failed",new Bundle());
+        initNotificationChannel();
         CleverTapAPI cleverTap = CleverTapAPI.getDefaultInstance(context);
-        CleverTapAPI.createNotificationChannel(context,clientId,clientId,"notification",NotificationManager.IMPORTANCE_MAX,true);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```
Comment (witcher-shailesh) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:320]: x-variable = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O ? NotificationManager.IMPORTANT_MAX : NotificationManager.IMPORTANCE_MAX;

CleverTapAPI.createNotificationChannel(context,clientId,"Promotion","Notifications Related to promotion",x-variable,true);

Comment (witcher-shailesh) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:320]: this looks cleaner

Comment (spravinkumar9952) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:320]: i checked that condintion for groupId parameter in `createNotificationChannel`. For that, we can't write like this.

Comment (spravinkumar9952) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:320]: because in older version there is no group id

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**